### PR TITLE
Tool 574 certificate helper

### PR DESCRIPTION
**Pull request by lpusok**

### CODE REVIEW

File: download_certificates.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	for i, p12 := range URLs {
+		log.Debugf("Downloading p12 file number %d from %s", i, p12.URL)
+		p12CertInfos, err := downloadPKCS12(p12.URL, p12.Passphrase)
+		if err != nil {
+			return nil, err
+		}
+
+		log.Debugf("Codesign identities included:")
+		for _, cert := range p12CertInfos {
+			log.Debugf("certificate Serial: %s, Name: %s, Team ID: %s, Team: %s, Expiration: %s", cert.Serial, cert.CommonName, cert.TeamID, cert.TeamName, cert.EndDate)
```
Comment (lpusok) [inline download_certificates.go]: call String()

File: download_certificates.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	}
+
+	// Local file
+	if url.Scheme == "file" {
+		src := strings.Replace(src, url.Scheme+"://", "", -1)
+
+		return ioutil.ReadFile(src)
+	}
+
+	// ToDo: add timeout, retry
```
Comment (lpusok) [inline download_certificates.go]: Add retry, timeout.

File: validate_certificates.go [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+	Invalid ProfileType = iota
+	Development
+	AdHoc
+	Enterprise
+	AppStore
+	Unsupported
+)
+
+// String returns a string representation of ProfileType
+func (p ProfileType) String() string {
```
Comment (lpusok) [inline validate_certificates.go]: Check network package if enums can be reused from there.

File: download_certificates.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
+package main
```
Comment (BirmacherAkos) [inline download_certificates.go]: Do not use underline in Golang file names.

Comment (BirmacherAkos) [reply download_certificates.go]: Merge the #68 and move the *download_certificates.go files under the `appstoreconnect` package.

Comment (BirmacherAkos) [reply download_certificates.go]: I would recommend moving the whole download_certificates.go file's content to the certificates.go

File: validate_certificates.go [hunk 5]
```diff
@@ -0,0 +1,226 @@
+package main
```
Comment (BirmacherAkos) [inline validate_certificates.go]: Do not use underline in Golang file names.

Comment (BirmacherAkos) [reply validate_certificates.go]: Merge the #68 and move the *validate_certificates.go files under the autoprovision package.

Comment (BirmacherAkos) [reply validate_certificates.go]: And please rename it to certificatehelper.go

File: autoprovision/certificatehelper.go [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+	for _, cert := range certs {
+		certInfo += cert.String() + "\n"
+	}
+	certInfo += "]"
+
+	return certInfo
+}
+
+// GetMatchingCertificates returns validated and matching with App Store Connect API certificates
+func GetMatchingCertificates(certificates []certificateutil.CertificateInfoModel, AppStoreConnectCertificates map[appstoreconnect.CertificateType][]AppStoreConnectCertificate, requiredCertificateTypes []appstoreconnect.CertificateType, typeToName map[appstoreconnect.CertificateType]string, teamID string) (map[appstoreconnect.CertificateType][]AppStoreConnectCertificate, error) {
```
[the remark this answers is not in the document]
Comment (BirmacherAkos) [reply autoprovision/certificatehelper.go]: If you go with my idea above (get the serial number, etc) then you don't need this method.

File: autoprovision/certificatehelper.go [hunk 7]
```diff
@@ -91,30 +92,42 @@ func queryAllIOSCertificates(client *appstoreconnect.Client) (map[CertificateTyp
...
-	})
-	if err != nil {
-		return nil, err
-	}
+	nextPageURL := ""
+	var responseCertificates []appstoreconnect.Certificate
+	for {
+		response, err := client.Provisioning.ListCertificates(&appstoreconnect.ListCertificatesOptions{
+			FilterCertificateType: certificateType,
+			Limit:                 10,
```
Comment (BirmacherAkos) [inline autoprovision/certificatehelper.go]: Why did you limit it to 10? 🤔

Comment (lpusok) [reply autoprovision/certificatehelper.go]: To have this code path used more often, avoiding breakage.

**Pull request merged**